core/src/database/operator.rs

```
@@ -386,7 +386,9 @@ impl Database {
         let query = sqlx::query(
             "
             INSERT INTO deposit_signatures (deposit_id, operator_xonly_pk, round_idx, kickoff_idx, kickoff_txid, signatures)
-            VALUES ($1, $2, $3, $4, $5, $6);"
+            VALUES ($1, $2, $3, $4, $5, $6)
+            ON CONFLICT (deposit_id, operator_xonly_pk, round_idx, kickoff_idx)
+            DO UPDATE SET kickoff_txid = EXCLUDED.kickoff_txid, signatures = EXCLUDED.signatures;"
```

was this causing a problem? also what does it mean if we try to override these signatures? will we lose the correct signatures when another deposit is triggered with the same ID?

---

This was causing an error about uniqueness when second deposit call is issued with same params. With this change, if a combination of (deposit_id, operator_xonly_pk, round_idx, kickoff_idx) is present in db, their (kickoff_txid, signatures) will be replaced with the latest call. Not sure if this is the correct behavior though; We can do nothing on conflict, too.

---

In this case it doesn't matter if we update or do nothing as both new and old signatures should be correct. I think in general also for get_params etc we should do nothing if it conflicts, so that for example if an operator somehow changes their kickoff wots keys we don't overwrite it in our db, because if we overwrite we cannot generate old signed kickoff txs. If an operator/verifier wants to change something about their xonly key/wots key, they need to create a new operator or verifier.

---

I think in terms of security measures we better re-sign the whole thing, since one can tweak some small details and if the verifiers use the same nonces again, they will compromise their private keys.

---

Than neither of the options should be viable, right? If altered `kickoff_txid` and `signatures` are given, we shouldn't silently ignore this conflict and throw an error, instead of doing nothing. This will require additional code, instead of just changing the query.

---

For each deposit operators share new wots keys (for our current code they will share same wots keys if deposit outpoint is the same but it can be changed by editing code). With different wots keys txids will change so if we use same nonce (idk if we do) and on conflict update wots keys in db, their keys will get leaked.

---

Can't we just add some randomness to verifiers' nonce generation?

---

https://github.com/chainwayxyz/clementine/pull/680/commits/52b9206c6413c6e97f603ea6db1aa232e6804fb3